aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,221 @@
 # -*- coding: utf-8 -*-
-"""`CalcJob` implementation for the pp.x code of Quantum ESPRESSO."""
+""" CalcJob implementation for pp.x """
 from __future__ import absolute_import
 
-from aiida.orm import RemoteData, FolderData
-from aiida_quantumespresso.calculations.namelists import NamelistsCalculation
+import os
+import six
+from six.moves import range
 
+from aiida import orm
+from aiida.common import datastructures, exceptions
 
-class PpCalculation(NamelistsCalculation):
+from aiida_quantumespresso.calculations import _lowercase_dict, _uppercase_dict
+from aiida_quantumespresso.utils.convert import convert_input_to_namelist_entry
+from .base import CalcJob
+
+
+def validate_parameters(value, ctx):  # pylint: disable=unused-argument
+    """ Validate 'parameters' dict."""
+
+    # Check for essential keys
+    try:
+        plot_num = value['INPUTPP']['plot_num']
```

Comment:
Also the `value` here will be the actual input value, so it is a `Dict` node. That means you should probably first do `parameters = value.get_dict()` and then do checks on that normal dictionary

Reply:
Fixed this, sorry. 
Added a test for the calculation class.